src/composables/storageRef.ts

```diff
@@ -51,20 +62,66 @@ export function useStorageRef<T = string | object | any[]>(
   }
 
   function setStorageState(val: T | null) {
-    WalletStorage.set(storageKey, (val && serializer?.write) ? serializer.write(val) : val);
+    storage?.set(storageKey, (val && serializer?.write) ? serializer.write(val) : val);
   }
 
   // Restore state and run watchers
   (async () => {
-    let restoredValue = await WalletStorage.get<T | null>(storageKey);
+    if (isSecure) {
```

Comment:
I think this composable should not be responsible of having logic of choosing and switching between different ways of storing data depending on the environment as this makes the code really hard to follow.

There is a `serializer` option for this composable that allows to perform actions like encoding/decoding every time the state is updated/set/read.